addons/repair/models/repair.py

```diff
@@ -89,9 +89,7 @@ def _default_picking_type_id(self):
         default=1.0, digits='Product Unit of Measure')
     product_uom = fields.Many2one(
         'uom.uom', 'Product Unit of Measure',
-        compute='compute_product_uom', store=True, precompute=True,
-        domain="[('category_id', '=', product_uom_category_id)]")
-    product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
+        compute='compute_product_uom', store=True, precompute=True)
```

Comment:
You need to set a domain here as well on the product's `allowed_uom_ids`, otherwise you can pick any UoM for a repair order.